examples/pbr.py

```diff
@@ -0,0 +1,249 @@
+import numpy as np
+import imageio
+import trimesh
+import wgpu
+import pygfx as gfx
+from pathlib import Path
+from wgpu.gui.auto import WgpuCanvas, run
+from pygfx.renderers.wgpu import Binding, WorldObjectShader, RenderMask
+
+
+class Skybox(gfx.WorldObject):
+    def __init__(self, material=None):
+        super().__init__(material=material)
+        self.box = gfx.box_geometry(2, 2, 2)
+
+
+class SkyboxMaterial(gfx.Material):
+    def __init__(self, *, map, **kwargs):
+        super().__init__(**kwargs)
+        self.map = map
+
+
+@gfx.renderers.wgpu.register_wgpu_render_function(Skybox, SkyboxMaterial)
+class SkyboxShader(WorldObjectShader):
```

Comment:
The builtin Background material already supports skyboxes. If that was not sufficient, we could extend the functuinality or maybe introduce a builtin Skybox material. Could be done in a separate PR though.

Reply:
I tried, but there seems to be a problem. The background is upside down.  I see a line of code in the `skybox.py` example:
https://github.com/pygfx/pygfx/blob/8375cbeaf5afd1a0a6957973af6f26ab7b3d16f6/examples/skybox.py#L43
It is used to flip the background, but so the objects in the scene also flip.

The implementation of skybox in this example is relatively correct. The only problem is that when rendering the skybox, we don't need to care about the depth test or write the depth buffer. The skybox should always be rendered first,  and put behind all objects in the scene. But now pygfx doesn't expose this ability (to control depth test and depth write), so I have to render skybox in another scene alone.

The same is true for stencil test and stencil write. I think we need to expose this capability so that we can achieve effects such as "outline".

Reply:
Ah, good point. I guess we need to add a property to the builtin lightbox to set the orientation. Good for a later PR. Let's discuss/track in #325.

At some point we could also make the renderer aware of background classes and render these first.